base_layer/common_types/src/emoji.rs

```diff
@@ -97,15 +94,15 @@ impl EmojiId {
 
     /// Try and construct an emoji ID from the given hex string. The method will fail if the hex is not a valid
     /// representation of a public key.
-    pub fn from_hex(hex_key: &str) -> Result<Self, HexError> {
+    pub fn from_hex(hex_key: &str) -> Result<Self, EmojiIdError> {
         let key = PublicKey::from_hex(hex_key)?;
         Ok(EmojiId::from_pubkey(&key))
     }
 
     /// Return the public key that this emoji ID represents
-    pub fn to_pubkey(&self) -> PublicKey {
-        let bytes = self.to_bytes();
-        PublicKey::from_bytes(&bytes).unwrap()
+    pub fn to_pubkey(&self) -> Result<PublicKey, EmojiIdError> {
+        let bytes = self.to_bytes()?;
```

Comment:
Although in principle I agree with avoiding unwraps, It's worth noting that this function can never fail as the code currently stands because there are 256 emojis and u8 ranges from 0 to 256.

Reply:
Ah nevermind, I see the input is a string - yes these unwraps are bad :)

Reply:
Actually, is see the `byte_vec` code is duplicated in `str_to_pubkey` and `str_to_pubkey` is used in the wallet.
So this cannot panic previously because you can only construct the String from bytes from a public key - this Struct can be greatly simplified :P